packages/next/server/edge-functions/response.ts

```diff
@@ -0,0 +1,264 @@
+          while (true) {
+            let { value, done } = await this.reader.read()
+            if (done) break
+            await this.writer.write(value)
+          }
+        }
+
+        this.writer.close()
+      })
+      .catch(console.error)
+  }
+
+  private get reader() {
+    if (!this._reader) {
+      this._reader = this._input.readable.getReader()
+    }
+
+    return this._reader
+  }
+
+  private get writer() {
+    if (!this._writer) {
+      this._writer = this._output.writable.getWriter()
+      this._streaming = true
+    }
+
+    return this._writer
+  }
+
+  private location(url: string) {
+    this.headers.set(
+      'Location',
+      url === 'back' ? this.headers.get('Referrer') || '/' : url
+    )
+  }
+
+  private _sendHeaders(event: HeadersEvent) {
+    if (this.headersSent) {
+      throw new Error(`Headers were already sent`)
+    }
+    this.headersSent = true
+    this._onHeadersSent(event, this)
+  }
+
+  public get body() {
+    return this._body
+  }
+
+  public get cookies() {
+    return this._cookieParser()
+  }
+
+  public get writable() {
+    return this._input.writable
+  }
+
+  public get readable() {
+    return this._output.readable
+  }
+
+  public status(statusCode: number) {
+    this.statusCode = statusCode
+    return this
+  }
+
+  public write(chunk: any) {
+    this.writer.write(typeof chunk === 'string' ? encode(chunk) : chunk)
+  }
+
+  public setHeaders(headers?: Record<string, string>) {
+    if (this.headersSent) {
+      throw new Error('Headers were sent')
+    }
+    for (let head in headers) {
+      this.headers.set(head, headers[head])
+    }
+
+    return this
+  }
+
+  public writeHead(code: number, headers?: Record<string, string>) {
+    this.statusCode = code
+    this.setHeaders(headers)
+    this._streaming = true
+    this._sendHeaders('streaming')
+  }
+
+  public cookie(
+    name: string,
+    value: { [key: string]: any } | string,
+    opts: CookieSerializeOptions = {}
+  ) {
+    const val =
+      typeof value === 'object' ? 'j:' + JSON.stringify(value) : String(value)
+
+    if (opts.maxAge) {
+      opts.expires = new Date(Date.now() + opts.maxAge)
+      opts.maxAge /= 1000
+    }
+
+    if (opts.path == null) {
+      opts.path = '/'
+    }
+
+    this.headers.append('Set-Cookie', cookie.serialize(name, String(val), opts))
+    return this
+  }
+
+  public clearCookie(name: string, opts: CookieSerializeOptions = {}) {
+    return this.cookie(name, '', { expires: new Date(1), path: '/', ...opts })
+  }
+
+  public json(obj: Dictionary): void {
+    const body = JSON.stringify(obj)
+    if (!this.headers.get('content-type')) {
+      this.headers.set('content-type', 'application/json')
+    }
+
+    this.send(body)
+  }
+
+  public send(
+    data:
+      | string
+      | number
+      | boolean
+      | Dictionary
+      | ReadableStream<Uint8Array>
+      | null = '',
+    headers?: Record<string, string>
+  ): void {
+    if (this.finished) {
```

Comment:
@javivelasco Made this update here to prevent sending additional body once the response has been finished.